src/browser/components/SelectCertificateModal.jsx

```diff
@@ -0,0 +1,162 @@
+import {Modal, Button, Table, Row, Col} from 'react-bootstrap';
+
+const CELL_SIZE = 23;
+const ELIPSIS_SIZE = 3;
+
+export default class SelectCertificateModal extends React.Component {
+  static propTypes = {
+    onSelect: PropTypes.func.isRequired,
+    onCancel: PropTypes.func,
+    certificateRequest: PropTypes.shape({
+      server: PropTypes.string,
+      certificateList: PropTypes.array,
+    }),
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedIndex: null,
+    };
+  }
+
+  maxSize = (item, max) => {
+    if (item.length <= max) {
+      return item;
+    }
+    const sub = item.substring(0, max - ELIPSIS_SIZE);
+    return `${sub}...`;
+  }
+
+  renderCert = (cert, index) => {
+    const issuer = cert.issuer && cert.issuer.commonName ? cert.issuer.commonName : '';
+    const subject = cert.subject && cert.subject.commonName ? cert.subject.commonName : '';
+    const serial = cert.serialNumber || '';
+
+    const issuerShort = this.maxSize(cert.issuer.commonName.split(':')[1], CELL_SIZE);
+    const subjectShort = this.maxSize(cert.subject.commonName.split(':')[1], CELL_SIZE);
+    const serialShort = this.maxSize(cert.serialNumber, CELL_SIZE);
+    const select = () => {
+      this.setState({selectedIndex: index});
+    };
+
+    const style = this.state.selectedIndex === index ? {background: '#457AB2', color: '#FFFFFF'} : {};
+    return (
+      <tr
+        key={`cert-${index}`}
+        onClick={select}
+        style={style}
+      >
+        <td
+          style={style}
+          title={issuer}
+        >{issuerShort}</td>
+        <td
+          style={style}
+          title={subject}
+        >{subjectShort}</td>
+        <td
+          style={style}
+          title={serial}
+        >{serialShort}</td>
+      </tr>);
+  };
+
+  renderCerts = (certificateList) => {
+    if (certificateList) {
+      const certs = certificateList.map(this.renderCert);
+      return (
+        <Fragment>
+          {certs}
+        </Fragment>
+      );
+    }
+    return (<Fragment><tr/><tr><td/><td>{'No certificates available'}</td><td/></tr></Fragment>);
+  }
+
+  getSelectedCert = () => {
+    return this.state.selectedIndex === null ? null : this.props.certificateRequest.certificateList[this.state.selectedIndex];
+  };
+
+  handleOk = () => {
+    const cert = this.getSelectedCert();
+    if (cert !== null) {
+      this.props.onSelect(cert);
+    }
+  }
+
+  handleCertificateInfo = () => {
+    const certificate = this.getSelectedCert();
+    if (certificate !== null) {
+      ipcRenderer.send('show-trusted-cert', certificate);
+    }
+  }
+
+  render() {
+    const certList = this.props.certificateRequest ? this.props.certificateRequest.certificateList : [];
+    const server = this.props.certificateRequest ? this.props.certificateRequest.server : '';
+    return (
+      <Modal
+        bsClass='modal'
+        className='certificateModal'
+        show={this.props.certificateRequest !== null}
```

Comment:
PR looking good!

This should be driven by state, yes?
`this.props.certificateRequest !== null`

Reply:
hey! sorry I didn't saw your comment, in this case it will disappear once the modal closes with either result (cert accepted/dennied)